python/rmm/rmm/_lib/device_buffer.pyx

```diff
@@ -138,6 +138,26 @@ cdef class DeviceBuffer:
         }
         return intf
 
+    def prefetch(self, device=None, stream=None):
```

Comment:
It feels slightly odd to have a free function in C++ but a method in Python. I guess this is more practically useful though? Also we can't have overloads in Python like you can in C++.

Reply:
There are no pointers in Python. RMM Python represents data allocations using DeviceBuffer. So it makes sense to only prefetch DeviceBuffers.

It could be a free function though, and then it could introspect the type to decide how to prefetch it (runtime overload, effectively).